### PR TITLE
Enable F402

### LINKED ISSUE #9295: Enable `F402`
F402 is diabled, but should be enabled. The following is a patch to enable the rule and fixes its violations:

```diff
diff --git a/mlflow/types/schema.py b/mlflow/types/schema.py
index 45addf974..3d7532b1e 100644
--- a/mlflow/types/schema.py
+++ b/mlflow/types/schema.py
@@ -119,11 +119,7 @@ class DataType(Enum):
 
     @classmethod
     def from_numpy_type(cls, np_type):
-        for dt in cls._member_map_.values():
-            if np_type == dt.to_numpy():
-                return dt
-
-        return None
+        return next((v for v in cls._member_map_.values() if v.to_numpy() == np_type), None)
 
 
 class ColSpec:
diff --git a/pyproject.toml b/pyproject.toml
index f78f8b07f..54d9e05d5 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -45,7 +45,6 @@ ignore = [
   "E501",  # line-too-long
   "E741",  # ambiguous-variable-name
   "F401",  # unused-import
-  "F402",  # import-shadowed-by-loop-var
   "F403",  # undefined-local-with-import-star
   "F811",  # redefined-while-unused
 ]
diff --git a/tests/autologging/test_autologging_utils.py b/tests/autologging/test_autologging_utils.py
index c98421410..ffd87967f 100644
--- a/tests/autologging/test_autologging_utils.py
+++ b/tests/autologging/test_autologging_utils.py
@@ -5,7 +5,6 @@ import time
 import sys
 import pytest
 from collections import namedtuple
-from unittest.mock import Mock, call
 from unittest import mock
 
 import mlflow
@@ -172,7 +171,7 @@ def test_wrap_patch_with_module():
 
 @pytest.fixture
 def logger():
-    return Mock()
+    return mock.Mock()
 
 
 def get_input_example():
@@ -196,8 +195,8 @@ def test_if_getting_input_example_fails(logger):
     assert input_example is None
     assert signature is None
     calls = [
-        call("Failed to gather input example: " + error_msg),
-        call(
+        mock.call("Failed to gather input example: " + error_msg),
+        mock.call(
             "Failed to infer model signature: "
             + "could not sample data to infer model signature: "
             + error_msg

```

### PR DESCRIPTION
**Pull request by zacharyftw**
## Related Issues/PRs

Resolve https://github.com/mlflow/mlflow/issues/9295

## What changes are proposed in this pull request?

F402 is diabled, but should be enabled. The following is a patch to enable the rule and fixes its violations.

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/autologging/test_autologging_utils.py [hunk 1]
```diff
@@ -196,8 +195,8 @@ def throws():
     assert input_example is None
     assert signature is None
     calls = [
-        call("Failed to gather input example: " + error_msg),
-        call(
+        mock.call("Failed to gather input example: " + error_msg),
++       mock.call(
```
Comment (harupy) [inline tests/autologging/test_autologging_utils.py:199]:
```suggestion
        mock.call(
```

File: mlflow/types/schema.py
Comment (harupy) [inline mlflow/types/schema.py:1]: /ai Is the new code equivalent to the old code?

**Review by harupy — APPROVED** [review]
LGTM!

**Comment by KekmaTime** [comment]
> LGTM!

hey  @harupy thanks ! btw what does the / ai command do

**Comment by harupy** [comment]
> btw what does the / ai command do

Good question! What it does is ask GPT4 a question.

File: pyproject.toml [hunk 2]
```diff
@@ -45,7 +45,6 @@ ignore = [
   "E402",  # module-import-not-at-top-of-file
   "E501",  # line-too-long
   "E741",  # ambiguous-variable-name
-  "F402",  # import-shadowed-by-loop-var
```
Comment (harupy) [inline pyproject.toml:48]: /ai Explain E402 in flake8

**Pull request merged**